app/src/lib/stores/updates/update-remote-url.ts

```diff
@@ -0,0 +1,29 @@
+import { Repository } from '../../../models/repository'
+import { IAPIRepository } from '../../api'
+import { GitStore } from '../git-store'
+
+export const updateRemoteUrl = async (
+  gitStore: GitStore,
+  repository: Repository,
+  apiRepo: IAPIRepository
+): Promise<void> => {
+  // I'm not sure when these early exit conditions would be met. But when they are
+  // we don't have enough information to continue so exit early!
+  if (!gitStore.currentRemote) {
+    return
+  }
+  if (!repository.gitHubRepository) {
+    return
+  }
+
+  const remoteUrl = gitStore.currentRemote.url
+  const updatedRemoteUrl = apiRepo.clone_url
+  const isProtocolHttps = remoteUrl && remoteUrl.startsWith('https://')
+  const usingDefaultRemote =
```

Comment:
I'm not quite following the logic here. Based on the naming I'd assume that this variable is supposed to hold a value indicating whether the current branch is tracking a branch on the default remote but what it seems to be checking for is merely that the current repository has a default remote and that the default remote is the same as the clone URL of the GitHubRepository which I think it'll always be since the GitHubRepository instance is assigned based on the default remote URL:

https://github.com/desktop/desktop/blob/d8665f8417b9d7b931d97f56ccaed88a139b383e/app/src/lib/stores/app-store.ts#L3167-L3169

I would expect the logic to look more like 👇 

```ts
const usingDefaultRemote =
  gitStore.defaultRemote &&
  gitStore.currentRemote.name === gitStore.defaultRemote.name
```

Reply:
we probably don't care about the `currentRemote` in this method since we're only interested in updating `origin`.